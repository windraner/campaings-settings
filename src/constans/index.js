export const MENU_LIST = [
  'Campaign details',
  'Media Collections',
  'Blacklists',
  'Media rights'
];

export const ACTIVE_TAB = 'activeTab';
export const NAME = 'name';
export const DESCRIPTION = 'description';
export const PRODUCT_CATALOGUE = 'productCatalogue';
export const IS_SHOPPABLE_LINK = 'isShoppableLink';
export const IS_MEDIA_COLLECTION = 'isMediaCollection';
export const MEDIA_COLLECTION_LIST = 'mediaCollectionList';
export const IS_PHOTO_MODERATION_QUEUE = 'isPhotosModerationQueue';
export const USER_BLACK_LIST = 'userBlackList';
export const HASHTAG_BLACK_LIST = 'hashtagBlackList';
export const ACCOUNTS_SETTINGS_ITEMS = 'accountsSettingsItems';
export const SHOW_ERROR_ALERT = 'showErrorAlert';
export const SHOW_SUCCESS_ALERT = 'showSuccessAlert';
export const IS_MEDIA_RIGHTS = 'isMediaRights';
export const HASHTAG_RIGHTS_LIST = 'hashtagRightsList';
export const TERMS_AND_CONDITIONS = 'termsAndConditions';

export const VALIDATED_FIELDS = [
  ACTIVE_TAB
];
