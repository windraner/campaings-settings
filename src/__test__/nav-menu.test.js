import React from 'react';
import NavMenu from '../components/main-content-section/nav-menu/nav-menu';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { MENU_LIST } from '../constans';

configure({ adapter: new Adapter() });

describe('<NavMenu />', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<NavMenu stateHandler={function() {}} activeTab={MENU_LIST[0]} menuAlerts={[]} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
