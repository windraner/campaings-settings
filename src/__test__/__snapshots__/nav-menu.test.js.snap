// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavMenu /> matches the snapshot 1`] = `
<ul
  className="campaigns-settings__nav-menu-list m-0 d-flex flex-column"
>
  <NavMenuItem
    activeTab="Campaign details"
    key="Campaign details"
    name="Campaign details"
    stateHandler={[Function]}
  />
  <NavMenuItem
    activeTab="Campaign details"
    key="Media Collections"
    name="Media Collections"
    stateHandler={[Function]}
  />
  <NavMenuItem
    activeTab="Campaign details"
    key="Blacklists"
    name="Blacklists"
    stateHandler={[Function]}
  />
  <NavMenuItem
    activeTab="Campaign details"
    key="Media rights"
    name="Media rights"
    stateHandler={[Function]}
  />
</ul>
`;
